vm/core/context.go

```diff
@@ -72,36 +134,80 @@ func (c *Context) Handler() Handler {
 }
 
 // Spawn account.
-func (c *Context) Spawn(args scale.Encodable) error {
-	account, err := c.load(ComputePrincipal(c.Header.TemplateAddress, args))
+func (c *Context) Spawn(template Address, blob []byte) (Address, error) {
+	// calculate new principal address
+	principalAddress := ComputePrincipalFromBlob(template, blob)
+
+	// check if the account is already spawned
+	account, err := c.load(principalAddress)
 	if err != nil {
-		return err
+		return Address{}, err
 	}
+	// the account is already spawned and contains different code. this should not happen.
+	// if len(account.State) > 0 && !bytes.Equal(account.State, blob) {
```

Comment:
should it be implemented? E.g
```suggestion
	// the account is already spawned and contains different code. this should not happen.
	if len(account.State) > 0 && !bytes.Equal(account.State, blob)
		panic("account is already spawned with different code")
	}
```